httpclient5/src/main/java/org/apache/hc/client5/http/config/RequestConfig.java

```diff
@@ -43,7 +43,6 @@
 public class RequestConfig implements Cloneable {
 
     private static final Timeout DEFAULT_CONNECTION_REQUEST_TIMEOUT = Timeout.ofMinutes(3);
-    private static final Timeout DEFAULT_CONNECT_TIMEOUT = Timeout.ofMinutes(3);
     private static final TimeValue DEFAULT_CONN_KEEP_ALIVE = TimeValue.ofMinutes(3);
```

Comment:
Another stupid question: `DEFAULT_CONN_KEEP_ALIVE` applies to the lifetime of the connection. Why is is retained here and not moved to the new config class?

Reply:
> 
> 
> Another stupid question: `DEFAULT_CONN_KEEP_ALIVE` applies to the lifetime of the connection. Why is is retained here and not moved to the new config class?

The `keep-alive` period is determined on a per request basis, generally based on `Keep-Alive` header.

Reply:
But this will affect the pool object after the request has been completed, doesn't it?

Reply:
@michael-o Yes, it would but this decision cannot be made by the connection management layer. It must be made by the HTTP protocol layer.

Reply:
So the overall idle time can change over time depending on this or the header value?

Reply:
Based on the request and response I think there are three results:
1. Do not keep the connection alive (`Connection: close` response header)
2. keep the connection alive for a specific amount of time `Keep-Alive: timeout=30`
3. keep the connection alive for an indeterminate amount of time so the default value can be used (http/1.1 without specifying close)

If memory serves, we use `null` for [1] (connection is not kept alive) and return a value based on either a keep-alive timeout value [2] or the default [3] when configured. If we can update the abstraction to differentiate states [1] from [3] I think the default configuration makes sense in the connection management configuration.

Reply:
@carterkozak To me, this is clearly an HTTP protocol layer responsibility, but if you, guys, are absolutely convinced I can move this setting to `ConnectionConfig`.

Reply:
I defer to your judgement, I could make a case for either design.

Reply:
I am accepting @ok2c's argumentation.